intel_adsp: Dedup Kconfig/DeviceTree Driver Options and Nodes

teburd opened the pull request:
There was a lot of copy paste going on (my own doing...) and I finally had enough. This tries to right the ship.

Commit:
soc/intel_adsp: Dedup the Kconfig driver options

The same options were copied to each Kconfig.defconfig, in different
places (my own doing) for both DMA and I2S. This dedups the defines and
moves them into the common soc Kconfig.defconfig.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>

MaureenHelm commented on soc/xtensa/intel_adsp/Kconfig.defconfig:
```diff
@@ -13,10 +13,14 @@ config XTENSA_CACHED_REGION
 config XTENSA_UNCACHED_REGION
 	default 4
 
+if SOC_FAMILY_INTEL_ADSP
+
```
This change should be in the previous commit, not the dts commit

Commit:
soc/intel_adsp: Dedup device tree nodes

The dma nodes in device tree were entirely copy pasted. Rather than
doing that lets create a common intel_cavs.dtsi each specialization
then includes. This dedups the lpgpdma entries.

Signed-off-by: Tom Burdick <thomas.burdick@intel.com>
